pkg/cgroup/cgroup_stats_linux.go

```diff
@@ -110,32 +112,28 @@ func (c CCgroupV1StatManager) SetCGroupStat(containerID string, cgroupStatMap ma
 }
 
 func (c CCgroupV2StatManager) SetCGroupStat(containerID string, cgroupStatMap map[string]*types.UInt64StatCollection) error {
-	stat, err := c.manager.Stat()
+	stat, err := c.manager.GetStats()
 	if err != nil {
 		return err
 	}
-	if stat.Memory == nil {
-		return fmt.Errorf("cgroup metrics does not exist, the cgroup might be deleted")
-	}
 	// memory
-	if stat.Memory != nil {
-		cgroupStatMap[config.CgroupfsMemory].SetAggrStat(containerID, stat.Memory.Usage)
-		cgroupStatMap[config.CgroupfsKernelMemory].SetAggrStat(containerID, stat.Memory.KernelStack)
-		cgroupStatMap[config.CgroupfsTCPMemory].SetAggrStat(containerID, stat.Memory.Sock)
-	}
+	cgroupStatMap[config.CgroupfsMemory].SetAggrStat(containerID, stat.MemoryStats.Usage.Usage)
```

Comment:
do you need to check if `stat.Memory` is `nil`?

Reply:
For runc, `stat.MemoryStat` is a struct and not a pointer to a struct so it will never be `nil`. However, I have tested `c.manager.GetStats()` with a nonexistent cgroup and an error such as the following will be returned: `error while statting cgroup v2: [openat2 /sys/fs/cgroup/nonexistentcgroup/cgroup.procs: no such file or directory]`